src/rust/engine/fs/store/src/lib.rs

```diff
@@ -787,17 +815,24 @@ impl Store {
       .to_boxed()
   }
 
-  fn materialize_file(
+  fn uncached_materialize_file(
     &self,
     destination: PathBuf,
-    digest: Digest,
-    is_executable: bool,
+    input: FileMaterializationInput,
     workunit_store: WorkUnitStore,
   ) -> BoxFuture<LoadMetadata, String> {
+    let FileMaterializationInput {
+      digest,
+      is_executable,
+    } = input;
+
     self
       .load_file_bytes_with(
         digest,
         move |bytes| {
+          // TODO: Determine when this situation happens! If we materialize files into a
```

Comment:
The store is used plenty of places other than for process execution. For instance, one may in a `@console_rule` use `Workspace.materialize_directory`.

We strongly assume that for process execution, because the caller has itself made an empty directory, we won't hit this codepath. I don't think it's worth adding an arg like `error_if_exists` to double-cover that case.